contributor-docs/adrs/adr-005-box-sx.md

```diff
@@ -0,0 +1,85 @@
+# ADR 005: Using Box for building components
+
+## Status
+
+Proposed
+
+## Context
+
+There are multiple ways to create components within primer/react and composing components from primer/react.
+
+The 2 popular methods are:
+
+1.  Creating components with styled-components
+
+    ```tsx
+    const Avatar = styled.img.attrs<StyledAvatarProps>(props => ({
+      height: props.size,
+      width: props.size
+    }))<StyledAvatarProps>`
+      display: inline-block;
+      overflow: hidden;
+      line-height: ${get('lineHeights.condensedUltra')};
+      border-radius: ${props => getBorderRadius(props)};
+      ${sx}
+    `
+    ```
+
+    [show full code example →](https://github.com/primer/react/pull/2019/files?diff=split&w=0)
+
+2.  Creating components with Box
+
+    ```tsx
+    const Avatar: React.FC<AvatarProps> = ({size = 20, alt = '', square = false, sx = {}, ...props}) => {
+      const styles = {
+        display: 'inline-block',
+        overflow: 'hidden',
+        lineHeight: 'condensedUltra',
+        borderRadius: getBorderRadius({size, square})
+      }
+
+      return (
+        <Box
+          as="img"
+          alt={alt}
+          sx={merge(styles, sx as SxProp)} // styles needs to merge with props.sx
+          {...props}
+        />
+      )
+    }
+    ```
+
+    [show full code example →](https://github.com/primer/react/pull/2019/files?diff=split&w=0)
+
+&nbsp;
+
+## Decision
+
+Prefer using method #2: Creating components with Box for the following reasons:
+
+- Better authoring experience with typescript. With Box, we can improve the API and autocomplete for consuming primitives.
+- The styling library used used is an implementation detail and we should be able to replace it. (Avoid leaky abstractions)
+- We have had issues with exporting types, we can increase confidence by keeping the exported types close to what we author.
+
+&nbsp;
+
+This conversation can be extended to overriding stlyes composing components and adding styles to them. We want to use the `sx` prop to add these styles.
```

Comment:
I'm not sure I understand what you're trying to say here.

Reply:
Going to remove this and move into an ADR of it's own!